### PR TITLE
Django 2 compatability

### PR DESCRIPTION
**Pull request by millerdev**
Poked at this a while ago, but didn't make a PR at the time. Sending along now since @dannyroberts is working on similar efforts (#26778).

### CODE REVIEW

File: corehq/middleware.py [hunk 1]
```diff
@@ -146,7 +146,7 @@ def process_view(self, request, view_func, view_args, view_kwargs):
             timeout = settings.SECURE_TIMEOUT
             # force re-authentication if the user has been logged in longer than the secure timeout
             if self._session_expired(timeout, request.user.last_login, now):
-                django_logout(request, template_name=settings.BASE_TEMPLATE)
+                LogoutView.as_view(template_name=settings.BASE_TEMPLATE)(request)
```
Comment (millerdev) [inline corehq/middleware.py]: I was unable to find clear instructions for how to replace `django_logout`, so I'm not certain that this is exactly equivalent.

Comment (snopoke) [reply corehq/middleware.py]: Seems to be right:
```
The LoginView and LogoutView class-based views supersede the deprecated login() and logout() function-based views.
```

Also looking at the code for `logout` it's doing the same thing as you are here.

**Review by snopoke — APPROVED** [review]

**Comment by dannyroberts** [comment]
I'm seeing

```
ImportError: cannot import name 'TokenType'
```

I wonder if there's no single Django 1 and 2 compatible import? Might have to have some logic that catches `ImportError` and pulls out the value from either `TOKEN_TEXT` or `TokenType.TEXT`, whichever one exists. If you're onboard, I'm happy to take over and add that change—I realize you were just putting up something you'd started on a while ago to help me out

**Review by millerdev — COMMENTED** [review]
👍 on latest changes

**Pull request merged**